File: InsertionSort1.py
import time

def insertionSort(arr):
    for i in range(1, len(arr)):
        key = arr[i]

        j = i - 1
        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key

if __name__ == "__main__":
	repeat = 10
	maxList = 20000
	r = 2000
	while r <= maxList:
		arr = []
		sum = 0
		for k in range(repeat):
			for i in range(r):
				arr.append(i)
			start = int(round(time.time() * 1000))
			insertionSort(arr)
			end = int(round(time.time() * 1000))
			sum = end - start
		print(f'{r}\t{sum}\t{sum/repeat}')
		arr.clear()
		r+=2000

File: InsertionSort1.cpp
#include <iostream>
#include<bits/stdc++.h>

using namespace std;

//int counter = 0;
void sort(int arr[], int n)
{
    int i, key, j; 
    for (i = 1; i < n; i++)
    { 
        key = arr[i]; 
        j = i - 1; 

        while (j >= 0 && arr[j] < key)
        { 
            arr[j + 1] = arr[j]; 
            j = j - 1; 
        } 
        arr[j + 1] = key; 
    } 
}

int main() {
	int repeat = 10;
    int maxlist = 20000;
    int range= 2000;
    int n, k, i;
    for (n = 2000; n <= maxlist; n+=range){
        int d[n];
        clock_t start_time;
        clock_t end_time1;
        clock_t result;
        double sum = 0;
        for (k=0; k<repeat; k++){
            for (i=n-1; i>=0; i--){
                d[i] = i;
//                cout << d[i] << endl;
//                cout << i<< endl;
            }
            int N = sizeof(d) / sizeof(d[0]); 
            start_time = clock();
            sort(d, N);
			end_time1 = clock();
			result = end_time1 - start_time;
			sum += result;
			
        }
        cout << n << "\t" << sum << "\t" << (double) sum/10<< endl;
    }
	return 0;
}
